samcli/lib/init/template_modifiers/xray_tracing_template_modifier.py

```diff
@@ -26,36 +29,58 @@ def _add_new_field_to_template(self):
         global_section_position = self._section_position(self.GLOBALS)
 
         if global_section_position >= 0:
-            function_section_position = self._section_position(self.FUNCTION, global_section_position)
+            self._add_tracing_to_function(global_section_position)
+            self._add_tracing_to_api(global_section_position)
+        else:
+            self._add_tracing_with_globals()
 
-            if function_section_position >= 0:
-                field_positon = self._field_position(function_section_position, self.FIELD_NAME)
-                if field_positon >= 0:
-                    self.template[field_positon] = self.TRACING
-                    return
+    def _add_tracing_with_globals(self):
+        resource_section_position = self._section_position(self.RESOURCE)
+        globals_section_data = [
+            self.COMMENT,
+            self.GLOBALS,
+            self.FUNCTION,
+            self.TRACING_FUNCTION,
+            self.API,
+            self.TRACING_API,
+            "\n",
+        ]
+        self.template = (
+                self.template[:resource_section_position]
+                + globals_section_data
+                + self.template[resource_section_position:]
+        )
 
-                new_fields = [self.TRACING]
-                _section_position = function_section_position
+    def _add_tracing_to_api(self, global_section_position):
```

Comment:
`_add_tracing_to_api` and `_add_tracing_to_function` are almost identical other than the field they looks for and replace. Could we make a single `_add_tracing_to_type(self, global_section_position, type, tracing_config_to_add)` and pass in the different properties from line 32 and 33?

Can you add typing to the function definition too?

Reply:
@mndeveci Reached out and looks like he had the opposite view. I will connect with him tomorrow during work hours and will respond back. Sorry for the back and forth on this, I didn't look at his comments but instead just went to the diff.

Reply:
Apologies for confusion, I've agreed what you said @jfuss, and addressed them with my latest commit.

Reply:
@jfuss 
Thank you for your review.
I agree with your advice.

And @mndeveci made a great fix.